pandas/core/generic.py

```diff
@@ -10648,15 +10649,16 @@ def _shift_with_freq(self, periods: int, axis: int, freq) -> Self:
                 raise ValueError(msg)
 
         elif isinstance(freq, str):
-            freq = to_offset(freq)
+            freq = to_offset(freq, is_period=True)
```

Comment:
is `is_period=True` always the case?

we may be missing test coverage here

for example, the following works with the latest release, but I think would fail here:
```python
In [1]: df = pd.DataFrame({'x': [1,2,3]}, index=pd.date_range('2000', periods=3))

In [2]: df
Out[2]:
            x
2000-01-01  1
2000-01-02  2
2000-01-03  3

In [3]: df.shift(freq='1MS')
Out[3]:
            x
2000-02-01  1
2000-02-01  2
2000-02-01  3
```

Can we make sure there's a test for this in a separate PR please?

Reply:
would it work to do `is_period=isinstance(index, PeriodIndex)`?